src/compiler/moduleNameResolver.ts

```diff
@@ -32,10 +32,27 @@ namespace ts {
      * Kinds of file that we are currently looking for.
      * Typically there is one pass with Extensions.TypeScript, then a second pass with Extensions.JavaScript.
      */
-    enum Extensions {
-        TypeScript, /** '.ts', '.tsx', or '.d.ts' */
-        JavaScript, /** '.js' or '.jsx' */
-        DtsOnly /** Only '.d.ts' */
+    const enum Extensions {

```

Comment:
Name this enum as `Extensions` seems misleading. I was expecting file extensions....though I have no good idea for alternatives 😞